### PR TITLE
FI-770 Process multi file outputs for one resource and pass empty Medication

### PR DESCRIPTION
**Pull request by yunwwang**
There are two changes in this Pull Request:
1) Handle multiple file output for a single FHIR resource.
2) Pass Medication validation test if Medication is not included in output.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Internal ticket for this PR:
- [x] Internal ticket links to this PR
- [x] Internal ticket is properly labeled (Community/Program)
- [x] Internal ticket has a justification for its Community/Program label
- [x] Code diff has been reviewed for extraneous/missing code
- [x] Tests are included and test edge cases
- [x] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: lib/modules/onc_program/bulk_data_group_validation_sequence.rb [hunk 1]
```diff
@@ -44,18 +44,32 @@ def test_output_against_profile(klass,
 
         lines_to_validate_parameter = get_lines_to_validate(bulk_lines_to_validate)
 
-        file = output.find { |item| item['type'] == klass }
+        file_list = output.find_all { |item| item['type'] == klass }
 
-        skip "Bulk Data Server export does not have #{klass} data" if file.nil?
+        if file_list.empty?
+          pass 'No Medication resource in output.' if klass == 'Medication'
```
Comment (arscan) [inline lib/modules/onc_program/bulk_data_group_validation_sequence.rb:50]:
```suggestion
          omit 'No Medication resources provided, and Medication resources are optional.' if klass == 'Medication'
```

File: lib/modules/onc_program/bulk_data_group_validation_sequence.rb [hunk 2]
```diff
@@ -44,18 +44,32 @@ def test_output_against_profile(klass,
...
 
-        success_count = check_file_request(file, klass, validate_all, lines_to_validate, must_supports)
+        success_count = 0
+
+        file_list.each do |file|
+          success_count += check_file_request(file, klass, validate_all, lines_to_validate, must_supports)
+        end
+
+        if success_count.zero? && (validate_all || lines_to_validate.positive?)
+          pass 'No Medication resource in output.' if klass == 'Medication'
```
Comment (arscan) [inline lib/modules/onc_program/bulk_data_group_validation_sequence.rb:69]:
```suggestion
          omit 'No Medication resources provided, and Medication resources are optional.' if klass == 'Medication'
```

File: lib/modules/onc_program/bulk_data_group_validation_sequence.rb [hunk 3]
```diff
@@ -44,18 +44,32 @@ def test_output_against_profile(klass,
...
 
-        skip "Bulk Data Server export for #{klass} is empty" if success_count.zero? && (validate_all || lines_to_validate.positive?)
+          skip "Bulk Data Server export for #{klass} is empty"
```
Comment (arscan) [inline lib/modules/onc_program/bulk_data_group_validation_sequence.rb:71]:
```suggestion
          skip "Bulk Data Server export did not provide any #{klass} resources."
```

File: lib/modules/onc_program/test/bulk_group_validation_sequence_test.rb [hunk 4]
```diff
@@ -606,6 +606,14 @@
 
       assert pass_exception.message == 'Successfully validated 0 resource(s).'
     end
+
+    it 'passes when no Medication export' do
+      pass_exception = assert_raises(Inferno::PassException) do
```
Comment (arscan) [inline lib/modules/onc_program/test/bulk_group_validation_sequence_test.rb:611]: This should be an omit, as my suggestions are above, so we check an OmitException if you accept that change request?

File: lib/modules/onc_program/bulk_data_group_validation_sequence.rb [hunk 5]
```diff
@@ -96,16 +108,16 @@ def check_file_request(file, klass, validate_all, lines_to_validate, must_suppor
         streamed_ndjson_get(file['url'], headers) do |response, resource|
           assert response.headers['Content-Type'] == 'application/fhir+ndjson', "Content type must be 'application/fhir+ndjson' but is '#{response.headers['Content-type']}"
 
-          break if !validate_all && line_count >= lines_to_validate && (klass != 'Patient' || @has_min_patient_count)
+          @patient_ids_seen = Set.new if klass == 'Patient' && @patient_ids_seen.nil?
```
Comment (arscan) [inline lib/modules/onc_program/bulk_data_group_validation_sequence.rb:111]: I don't really understand this and the other `@patient_ids_seen` initialization.  Can't we just do
```
@patient_ids_seen = Set.new
```
in the `initialize` method?  Do we need to be resetting it along the way?

File: lib/modules/onc_program/bulk_data_group_validation_sequence.rb (hunk 3 above)
Comment (arscan) [reply lib/modules/onc_program/bulk_data_group_validation_sequence.rb:71]: oh shoot something weird happened.  The point here is that we should use `omit` not skip if medication.

File: lib/modules/onc_program/bulk_data_group_validation_sequence.rb (hunk 2 above)
Comment (yunwwang) [reply lib/modules/onc_program/bulk_data_group_validation_sequence.rb:69]: fixed

File: lib/modules/onc_program/test/bulk_group_validation_sequence_test.rb (hunk 4 above)
Comment (yunwwang) [reply lib/modules/onc_program/test/bulk_group_validation_sequence_test.rb:611]: fixed

File: lib/modules/onc_program/bulk_data_group_validation_sequence.rb (hunk 5 above)
Comment (yunwwang) [reply lib/modules/onc_program/bulk_data_group_validation_sequence.rb:111]: fixed

File: lib/modules/onc_program/bulk_data_group_validation_sequence.rb (hunk 3 above)
Comment (yunwwang) [reply lib/modules/onc_program/bulk_data_group_validation_sequence.rb:71]: fixed

**Pull request merged**